ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlUserContextProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.security;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+
+import io.confluent.ksql.rest.server.context.ConfiguredKafkaClientSupplier;
+import java.security.Principal;
+import java.util.function.Supplier;
+
+/**
+ * Provides access to clients required to communicate with remote services using the context of
+ * a specified user name and/or credentials.
+ * <p/>
+ * This context is used by KSQL to access Kafka and/or Schema Registry resources when a user
+ * executes a KSQL command so it can access those resources using the same user's permissions.
+ */
+public interface KsqlUserContextProvider {
+  /**
+   * Constructs a {@link ConfiguredKafkaClientSupplier} to access Kafka resources on the
+   * context of the specified user {@code principal}.
+   * </p>
+   * {@code Note:} The {@code ConfiguredKafkaClientSupplier} is required by Websockets
```

Comment:
Why are we explicitly calling out that `ConfiguredKafkaClientSupplier` is used by the WebSocket endpoint? The REST pathway also uses `ConfiguredKafkaClientSupplier`, does it not?

Reply:
The problem was on the `PrintTopicUtil` which it checks if the `KafkaClientSupplier` is an instance of `ConfiguredKafkaClientSupplier` in order to impersonate the PRINT command.

It should not be required to return a `ConfiguredKafkaClientSupplier`, as REST will use whatever `KafkaClientSupplier` is returned, and it is up to the user to return any implementation of a `KafkaClientSupplier` as long as it uses the User context. However, because WS uses this specific client to create an impersonated consumer, then I had to specify it in the method.

But you're right on the comment, perhaps I just need to call out that also REST uses this `ConfiguredKafkaClientSupplier` (not just WS). That removes implementation details from the javadoc.